tools/mot_visualization.py

```diff
@@ -0,0 +1,227 @@
+         '/ save the results) with the argument "--out-dir" or "--show"')
+
+    if not args.result_file.endswith(('.pkl', 'pickle')):
+        raise ValueError('The result file must be a txt file.')
+
+    # define output
+    if args.output is not None:
+        if args.out_video:
+            out_dir = tempfile.TemporaryDirectory()
+            out_path = out_dir.name
+            _out = args.output.rsplit('/', 1)
+            if len(_out) > 1:
+                os.makedirs(_out[0], exist_ok=True)
+        else:
+            out_path = args.output
+            os.makedirs(out_path, exist_ok=True)
+
+    cfg = Config.fromfile(args.config)
+    dataset = build_dataset(cfg.data.val, dict(test_mode=True))
+    results = mmcv.load(args.result_file)
+
+    # format the results to txts
+    resfiles, names, tmp_dir = dataset.format_results(results, None, ['track'])
+
+    for name in names:
+        print(f'Start processing video {name}')
+        if 'half-train' in dataset.ann_file:
+            gt_file = osp.join(dataset.img_prefix,
+                               f'{name}/gt/gt_half-train.txt')
+        elif 'half-val' in dataset.ann_file:
+            gt_file = osp.join(dataset.img_prefix,
+                               f'{name}/gt/gt_half-val.txt')
+        else:
+            gt_file = osp.join(dataset.img_prefix, f'{name}/gt/gt.txt')
+        res_file = osp.join(resfiles['track'], f'{name}.txt')
+        gt = mm.io.loadtxt(gt_file)
+        res = mm.io.loadtxt(res_file)
+        ini_file = osp.join(dataset.img_prefix, f'{name}/seqinfo.ini')
+        if osp.exists(ini_file):
+            acc, ana = mm.utils.CLEAR_MOT_M(gt, res, ini_file)
+        else:
+            acc = mm.utils.compare_to_groundtruth(gt, res)
+
+        frame_id_list = list(set(acc.mot_events.index.get_level_values(0)))
+        for frame_id in frame_id_list:
+            # events in the current frame
+            events = acc.mot_events.xs(frame_id)
+            cur_res = res.loc[frame_id] if frame_id in res.index else None
+            cur_gt = gt.loc[frame_id] if frame_id in gt.index else None
+            # path of img
+            img = osp.join(dataset.img_prefix,
+                           f'{name}/img1/{frame_id:06d}.jpg')
+            fps = events[events.Type == 'FP']
+            fns = events[events.Type == 'MISS']
+            idsws = events[events.Type == 'SWITCH']
+
+            bboxes, ids, wrong_types = [], [], []
+            for fp_index in fps.index:
+                hid = events.loc[fp_index].HId
+                bboxes.append([
+                    cur_res.loc[hid].X, cur_res.loc[hid].Y,
+                    cur_res.loc[hid].X + cur_res.loc[hid].Width,
+                    cur_res.loc[hid].Y + cur_res.loc[hid].Height,
+                    cur_res.loc[hid].Confidence
+                ])
+                ids.append(hid)
+                wrong_types.append(0)
+            for fn_index in fns.index:
```

Comment:
add comment to explain what `wrong_type` stand for. 
For example, `0 denotes false positive error`